docs/content/migrate/nginx-to-traefik.md

```diff
@@ -0,0 +1,726 @@
+- TLS secrets must stay in the same namespace as the Ingress and list the hostnames you serve.
+- NGINX `ssl-redirect` / `force-ssl-redirect` annotations are honored (unless globally forced).
+- After switching, verify with `kubectl describe ingress <name>` and `curl -k https://<host>/`.
+
+---
+
+## Step 6: Monitor and Validate
+
+After migrating Ingress resources, monitor traffic to ensure everything is working correctly.
+
+### Validation Commands
+
+```bash
+# Check Ingress resources are discovered by Traefik
+kubectl get ingress --all-namespaces
+
+# View Traefik logs for routing decisions
+kubectl logs -n traefik deployment/traefik -f
+
+# Test application endpoints
+curl -v https://myapp.example.com/api/health
+```
+
+### Common Issues and Troubleshooting
+
+**Issue: Ingresses not discovered by Traefik**
+
+Check IngressClass configuration:
+
+```bash
+# Verify IngressClass exists
+kubectl get ingressclass
+
+# Check Traefik is watching the correct IngressClass
+kubectl logs -n traefik deployment/traefik | grep -i "ingress"
+```
+
+**Issue: Annotation not working as expected**
+
+Some NGINX annotations have behavioral differences. Check the [limitations documentation](../reference/routing-configuration/kubernetes/ingress-nginx.md#limitations) for details.
+
+**Issue: TLS certificates not working**
+
+Ensure TLS secrets are in the same namespace as the Ingress:
+
+```bash
+# List secrets in the namespace
+kubectl get secrets -n <namespace>
+
+# Check TLS secret format
+kubectl get secret <tls-secret-name> -n <namespace> -o yaml
+```
+
+---
+
+## Step 7: Decommission NGINX Ingress Controller
+
+Once all traffic has been successfully migrated and validated, you can safely remove the NGINX Ingress Controller.
+
+### Verification Checklist
+
+Before decommissioning NGINX, ensure:
+
+- All Ingresses are routing through Traefik
+- No errors in application logs
+- Monitoring shows healthy traffic patterns
+- TLS certificates are working correctly
+- Authentication and authorization are functioning
+- NGINX admission webhook is no longer needed (Traefik is the only Ingress controller)
+
+### Remove NGINX Resources
+
+!!! info
+    
+    In this section, we assume that the NGINX Ingress Controller was installed in the namespace `ingress-nginx`. If it was installed in a different namespace, you need to replace `ingress-nginx` with the namespace name.
+  
+If NGINX Ingress Controller was installed via Helm, run the following command to uninstall the chart and clean up chart-managed resources:
+
+```bash
+helm uninstall ingress-nginx -n ingress-nginx
+```
+
+!!! warning "IngressClass Deleted by Helm"
+
+    When you run `helm uninstall`, the `nginx` IngressClass is also deleted. If you followed the Helm values configuration in [Step 1](#step-1-install-traefik-with-helm) (which includes `extraObjects` to create the IngressClass), Traefik will continue to work. If you did not include the `extraObjects` section, you must manually create the IngressClass:
```

Comment:
Since the extra objects create the same ingress class, wouldn't the Nginx uninstall delete that one, too? If there was a conflict and the nginx-originated one was deleted to add the Traefik one, I think this one will also be removed upon uninstalling nginx. So in the end, the ingress class has to be added back, even if you had it in extra objects.